python/ray/train/tests/test_batch_predictor.py

```diff
@@ -154,6 +177,98 @@ def test_batch_prediction():
     ]
 
 
+def test_batch_prediction_various_combination():
+    """Dataset level predictor test against various
+    - Dataset format
+    - Preprocessor implementation (pandas vs numpy)
+    - Predictor implementation (pandas vs numpy)
+    """
+    # Got to inline this rather than using @pytest.mark.parametrize to void
+    # unknown object owner error when running test with python cli.
+    test_cases = [
+        (
+            DummyPreprocessor(),
+            DummyPredictor,
+            ray.data.from_pandas(pd.DataFrame({"x": [1, 2, 3]})),
+            # Pandas dataset format should be preserved in output.
+            "pandas",
+        ),
+        (
+            DummyWithNumpyPreprocessor(),
+            DummyPredictor,
+            ray.data.from_pandas(pd.DataFrame({"x": [1, 2, 3]})),
+            # Pandas dataset format should be preserved in output.
+            "pandas",
+        ),
+        (
+            DummyPreprocessor(),
+            DummyWithNumpyPredictor,
+            ray.data.from_pandas(pd.DataFrame({"x": [1, 2, 3]})),
+            # Pandas dataset format should be preserved in output.
+            "pandas",
+        ),
+        (
+            DummyWithNumpyPreprocessor(),
+            DummyWithNumpyPredictor,
+            ray.data.from_pandas(pd.DataFrame({"x": [1, 2, 3]})),
+            # Preprocessing and prediction is done in numpy, but preprocessor
+            # batch format is still pandas, thus output is casted back.
+            "pandas",
+        ),
+        (
+            DummyPreprocessor(),
+            DummyWithNumpyPredictor,
+            ray.data.from_arrow(pa.Table.from_pydict({"x": [1, 2, 3]})),
+            # Output dataset format should be pandas given only pandas
+            # preprocessor is available.
+            "pandas",
+        ),
+        (
+            DummyWithNumpyPreprocessor(),
+            DummyWithNumpyPredictor,
+            ray.data.from_arrow(pa.Table.from_pydict({"x": [1, 2, 3]})),
+            # Output dataset format should be arrow given numpy path is taken
+            # for both preprocessor and predictor.
+            "arrow",
+        ),
+        (
+            DummyPreprocessor(),
+            DummyPredictor,
+            ray.data.from_arrow(pa.Table.from_pydict({"x": [1, 2, 3]})),
+            # Output dataset format should be pandas given only pandas
+            # preprocessor is available.
+            "pandas",
```

Comment:
IMO I think we should remove these output dataset format checks. It gets too much into the implementation details and makes maintenance of these tests difficult.

Ultimately, from the user perspective, the underlying dataset format of the output dataset does not matter

Reply:
this is for asserting our internal behavior and not exposed to end users right ?

Reply:
discussed offline to keep it asserting output formats working as expected